go/vt/vtgate/planbuilder/operators/fk_verify.go

```diff
@@ -0,0 +1,140 @@
+/*
```

Comment:
I suggest we wait with adding this operator until we have a clearer idea of what the engine primitives will look like

Reply:
As discussed offline, the plan is to break to insert primitive into smaller composable primitives to make adding new primitives similar and easier to follow.